### PR TITLE
Introduced a test suite for the demo tool

### LINKED ISSUE #78: Implement a test suite for the demo tool
Currently, there is very few tests for the demo tool. We need to add more tests to verify if it works fine. For example, the test suite may be based on shell scripts. Ideally, the new test suite should be integrated to the build process. However, if it takes long, it may be run separately (in this case, the docs need to be updated to remind contributors to run the test suite).

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of main updates:

- Added several shell-based tests for the demo tool.
- Added a README file with instructions.
- Updated the contributing guideline (there is a small update for commit messages).

This closes #78

### CODE REVIEW

File: src/test/shell/tool/github/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+```
+TOKEN=xyz # put your token for GitHub
+export TOKEN
+bash src/test/shell/tool/github/run_tests.sh
+```
+
+Logs are going to be available in `src/test/shell/tool/github`.
+
+When running first time, the tests take longer.
```
Comment (sourabhsparkala) [inline src/test/shell/tool/github/README.md:22]: When running for the first time, the tests may take longer to execute.

File: src/test/shell/tool/github/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Next time, the tests will use cached data which speed them up.
```
Comment (sourabhsparkala) [inline src/test/shell/tool/github/README.md:23]: During the next run, the tests will use the cached data, which in turn, speeds up the test run.

File: src/test/shell/tool/github/README.md (hunk 1 above)
Comment (artem-smotrakov) [reply src/test/shell/tool/github/README.md:22]: I'll add "for the" but the tests really take longer to execute.

File: src/test/shell/tool/github/README.md (hunk 2 above)
Comment (artem-smotrakov) [reply src/test/shell/tool/github/README.md:23]: Well, I'll add "the" but the rest looks fine to me.

**Comment by sourabhsparkala** [comment]
I guess this PR needs to be rebased.

**Comment by artem-smotrakov** [comment]
> I guess this PR needs to be rebased.

Yeah, just did it.

**Comment by artem-smotrakov** [comment]
Thanks for the review @sourabhsparkala !

**Pull request merged**